roles/kubernetes-control-plane/tasks/main.yml

```
@@ -1,3 +1,175 @@
+- name: Pre configure control plane hooks
+  ansible.builtin.include_tasks: "{{ item }}"
+  with_items: "{{ kubernetes_hookfiles.pre_configure_control_planes }}"
+  when:
+  - kubernetes_hookfiles.pre_configure_control_planes is defined
+
+- name: Create kubernetes/config/patches directory
+  ansible.builtin.file:
+    dest: "{{ item }}"
+    state: directory
+    mode: '0700'
+  with_items:
+  - "{{ kubernetes_config_directory }}/patches"
+  - "{{ kubernetes_scripts_directory }}"
+
+- name: Create kubernetes configuration files
+  ansible.builtin.template:
+    src: "{{ item.file }}"
+    dest: "{{ item.destination }}"
+    owner: root
+    group: root
+    mode: "{{ item.mode | default('0600') }}"
+    backup: false
+  with_items:
+  - {file: "encryption-{{ kubernetes_version }}.yaml.j2", destination: "{{ kubernetes_config_directory }}/encryption.yaml"}
+  # CIS 3.1.1, 3.1.2, 3.1.3
+  - {file: "kube-api-authentication-{{ kubernetes_version }}.yaml.j2", destination: "{{ kubernetes_config_directory }}/kube-api-authentication.yaml"}
+  # CIS 1.2.6, 1.2.7, 1.2.8
+  # STIG V-242382
+  - {file: "kube-api-authorization-{{ kubernetes_version }}.yaml.j2", destination: "{{ kubernetes_config_directory }}/kube-api-authorization.yaml"}
+  # STIG V-254800
+  - {file: "admission-configuration-{{ kubernetes_version }}.yaml.j2", destination: "{{ kubernetes_config_directory }}/admission-configuration.yaml"}
+  # CIS 1.1.12
+  # STIG V-242459
+  - {file: "etcd-patch-{{ kubernetes_version }}.yaml.j2", destination: "{{ kubernetes_config_directory }}/patches/etcd.yaml"}
+
+- name: Create kubernetes automatic certificate renewal scripts and service files
+  ansible.builtin.template:
+    src: "{{ item.file }}"
+    dest: "{{ item.destination }}"
+    owner: root
+    group: root
+    mode: "{{ item.mode | default('0600') }}"
+    backup: false
+  when: kubernetes_manage_cert_renewal
+  with_items:
+  - {file: "k8s-certs-renew.sh.j2", destination: "{{ kubernetes_scripts_directory }}/k8s-certs-renew.sh", mode: "0755"}
+  - {file: "k8s-certs-renew.service.j2", destination: "/etc/systemd/system/k8s-certs-renew.service", mode: "0644"}
+  - {file: "k8s-certs-renew.timer.j2", destination: "/etc/systemd/system/k8s-certs-renew.timer", mode: "0644"}
+
+- name: Copy additional control plane files
+  ansible.builtin.copy:
+    src: "{{ item.source }}"
+    dest: "{{ item.destination }}"
+    owner: root
+    group: root
+    mode: "{{ item.mode | default('0600') }}"
+  with_items: "{{ kubernetes_control_plane_additional_files }}"
```

Change item to something unique to avoid warning/duplicating use of item.
```suggestion
- name: Copy additional control plane files
  loop: "{{ kubernetes_control_plane_additional_files }}"
  loop_control:
    loop_var: additional_control_plane_file
  ansible.builtin.copy:
    src: "{{ additional_control_plane_file.source }}"
    dest: "{{ additional_control_plane_file.destination }}"
    owner: root
    group: root
    mode: "{{ additional_control_plane_file.mode | default('0600') }}"
```